### PR TITLE
New mass generate

### PR DESCRIPTION
**Pull request by atm-maxime**
Allow to mass generate PDF document from lists

### CODE REVIEW

File: htdocs/core/actions_massactions.inc.php [hunk 1]
```diff
@@ -1130,6 +1130,67 @@
...
+				$outputlangs->setDefaultLang($newlang);
+			}
+	
+			// To be sure vars is defined
+			if (empty($hidedetails)) $hidedetails=0;
+			if (empty($hidedesc)) $hidedesc=0;
+			if (empty($hideref)) $hideref=0;
+			if (empty($moreparams)) $moreparams=null;
+			
+			$result= $objecttmp->generateDocument($objecttmp->modelpdf, $outputlangs, $hidedetails, $hidedesc, $hideref, $moreparams);
```
Comment (eldy) [inline htdocs/core/actions_massactions.inc.php:1203]: What happen if $objecttmp->modelpdf is not defined ?

Comment (atm-maxime) [reply htdocs/core/actions_massactions.inc.php:1203]: Every generateDocument function checks that and use the default pdf model if ! dol_strlen($modele)

**Comment by eldy** [comment]
So my understanding is that this feature is not "Generate doc" but 
"Regenerate doc if already exisintg". Am i right ?
If yes, we should find a label into combo box so this will be clear for user to. I suggest 
"Regenerate existing main documents"

**Comment by atm-maxime** [comment]
I called it "Generate" because it's just doing massively what I could do by going on each object and clicking on Generate in the attached file box on the bottom left. It uses the modelpdf already saved  on the object. It generates whether the PDF is already existing or not. I just kept the same wording.
Also, next improvement will be to add the model selection before confirming generation.

**Pull request merged**